packages/dd-trace/src/aiguard/sdk.js

```diff
@@ -0,0 +1,172 @@
+'use strict'
+
+const NoopAIGuard = require('./noop')
+const executeRequest = require('./client')
+const {
+  AI_GUARD_RESOURCE,
+  AI_GUARD_TARGET_TAG_KEY,
+  AI_GUARD_REASON_TAG_KEY,
+  AI_GUARD_ACTION_TAG_KEY,
+  AI_GUARD_BLOCKED_TAG_KEY,
+  AI_GUARD_META_STRUCT_KEY,
+  AI_GUARD_TOOL_NAME_TAG_KEY
+} = require('./tags')
+const log = require('../log')
+const { URL } = require('url')
+
+const ALLOW = 'ALLOW'
+
+class AIGuardAbortError extends Error {
+  constructor (reason) {
+    super(reason)
+    this.name = 'AIGuardAbortError'
+    this.reason = reason
+  }
+}
+
+class AIGuardClientError extends Error {
+  constructor (message, opts = {}) {
+    super(message)
+    this.name = 'AIGuardClientError'
+    if (opts.errors) {
+      this.errors = opts.errors
+    }
+    if (opts.cause) {
+      this.cause = opts.cause
+    }
+  }
+}
+
+class AIGuard extends NoopAIGuard {
+  constructor (tracer, config) {
+    super(tracer)
+
+    if (!config.apiKey || !config.appKey) {
+      const message = 'AIGuard: missing api and/or app keys, use env DD_API_KEY and DD_APP_KEY'
+      log.error(message)
+      throw new Error(message)
+    }
+    this._headers = {
+      'DD-API-KEY': config.apiKey,
+      'DD-APPLICATION-KEY': config.appKey,
+    }
+
+    if (config.protocolVersion !== '0.4') {
+      log.error('AIGuard: observability of evaluation results requires protocol version 0.4')
+    }
+
+    let endpoint = config.aiguard?.endpoint
+    if (!endpoint) {
+      endpoint = `https://app.${config.site}/api/v2/ai-guard`
+    }
+    this._evaluateUrl = new URL(`${endpoint}/evaluate`)
+
+    this._timeout = config.aiguard.timeout
+    this._maxMessagesLength = config.experimental.aiguard.maxMessagesLength
+    this._maxContentSize = config.experimental.aiguard.maxContentSize
+    this._meta = { service: config.service, env: config.env }
+  }
+
+  _truncate (messages) {
+    const size = Math.min(messages.length, this._maxMessagesLength)
+    const result = new Array(size)
+    for (let i = 0; i < size; i++) {
+      let message = messages[i]
+      if ('content' in message && message.content.length > this._maxContentSize) {
+        message = { ...message }
+        message.content = message.content.slice(0, this._maxContentSize)
+      }
+      result[i] = message
+    }
+    return result
```

Comment:
```suggestion
    const result = messages.slice(0, size)
    
    for (let i = 0; i < size; i++) {
      const message = result[i]
      if ('content' in message && message.content.length > this._maxContentSize) {
        result[i] = { ...message, content: message.content.slice(0, max) };
      }
    }
    return result
```

I think this is faster with slice instead of manual copy

Reply:
Done, thanks!